src/FuseGPUThreadLoops.cpp

```diff
@@ -761,11 +761,13 @@ class ExtractSharedAndHeapAllocations : public IRMutator {
                                 return IRMutator::visit(op);
                             }
                         }
-                        const string &alloc_name;
+                        const string alloc_name;
                         const string &group_name;
 
                     public:
-                        RewriteGroupAccess(const string &alloc_name, const string &group_name)
+                        // Because we're passed in a temporary to construct this, don't capture
+                        // alloc_name by reference
+                        RewriteGroupAccess(const string alloc_name, const string &group_name)
```

Comment:
Doesn't this make an extra copy now? I think clang-tidy might complain unless you throw a std::move in the initialization. Alternatively, the argument type could remain a const ref.